app/controllers/users_controller.rb

```diff
@@ -1,13 +1,15 @@
 class UsersController < ApplicationController
   before_action :authenticate_user!
 
-  # GET /users or /users.json
   def index
     @users = User.all
   end
 
-  # GET /users/1 or /users/1.json
   def show
+    @user = User.find(params[:id])
+    @follow = User.where(user_id: current_user.id)
+    @posts = Post.where(id: @user.id)
+    @user_likes = @user.likes.where(post_id: @posts.map(&:id)).index_by(&:post_id)
```

Comment:
You could use `Post.where(id: @user.id).pluck(:id)` and save it directly in the `@posts` variable instead of get the posts and then get the ids with `@posts.map(&:id)`

This is how it would look like:
```ruby
@posts_ids = Post.where(id: @user.id).pluck(:id)
@user_likes = @user.likes.where(post_id: @posts_ids).index_by(&:post_id)
```

Use pluck instead of map, in many cases, it could show an increase in performance
#
You can read more about pluck vs map in this articles:
- https://www.fastruby.io/blog/performance/rails/writing-fast-rails.html
- https://www.rubyinrails.com/2014/06/05/rails-pluck-vs-select-map-collect/